numpy/core/tests/test_numeric.py

```diff
@@ -2351,44 +2357,29 @@ def test_clip_property(self, data, shape):
         That accounts for most of the function; the actual test is just three
         lines to calculate and compare actual vs expected results!
         """
-        # Our base array and bounds should not need to be of the same type as
-        # long as they are all compatible - so we allow any int or float type.
-        dtype_strategy = hynp.integer_dtypes() | hynp.floating_dtypes()
-
-        # The following line is a total hack to disable the varied-dtypes
-        # component of this test, because result != expected if dtypes can vary.
-        dtype_strategy = st.just(data.draw(dtype_strategy))
-
-        # Generate an arbitrary array of the chosen shape and dtype
-        # This is the value that we clip.
-        arr = data.draw(hynp.arrays(dtype=dtype_strategy, shape=shape))
-
         # Generate shapes for the bounds which can be broadcast with each other
         # and with the base shape.  Below, we might decide to use scalar bounds,
         # but it's clearer to generate these shapes unconditionally in advance.
         in_shapes, result_shape = data.draw(
             hynp.mutually_broadcastable_shapes(
                 num_shapes=2,
-                base_shape=shape,
+                base_shape=arr.shape,
                 # Commenting out the min_dims line allows zero-dimensional arrays,
                 # and zero-dimensional arrays containing NaN make the test fail.
                 min_dims=1
-  
             )
         )
+        # This test can fail if we allow either bound to be a scalar `nan`, or
+        # if bounds are of a different (still integer or float) dtype than the 
+        # array.  At some point we should investigate and fix those problems.
```

Comment:
This sounds like a bug. If there is not an open issue, it might be good to open one, and mention there that this was discovered by hypothesis.

Reply:
The `nan` thing is known and part of a deprecation pathway; I'll open an issue for the dtype inconsistency though.